### PR TITLE
Add shared memory to cgroup

### PR DESCRIPTION
**Pull request by thiagoftsm**
##### Summary
This is the second step to bring integration between eBPF and cgroups https://github.com/netdata/netdata/issues/11558.

This PR is creating shared memory with cgroup plugin.
##### Component Name
cgroup
ebpf.plugin
##### Test Plan

1 - Enable flag inside your `netdata.conf`
2 - Compile this branch
3 - Take a look in the logs to verify whether shared memory was filled

##### Additional Information
This PR is part of a huge branch that has almost all modifications proposed for #11558 , and it was tested on:

- Manjaro 21.1 (cgroup version 2)
- CentOS 7.9
- Debian 10.10
- Ubuntu 18.04

### CODE REVIEW

File: collectors/cgroups.plugin/sys_fs_cgroup.c [hunk 1]
```diff
@@ -461,6 +466,61 @@ void read_cgroup_plugin_configuration() {
...
+                                              cgroup_enable_systemd_services_detailed_memory |
+                                              cgroup_used_memory;
+    shm_cgroup_ebpf.header->body_length = length;
+
+    sem_post(shm_mutex_cgroup_ebpf);
+}
+
+void netdata_cgroup_ebpf_initialize_shm()
+{
+    shm_fd_cgroup_ebpf = shm_open(NETDATA_SHARED_MEMORY_EBPF_CGROUP_NAME, O_CREAT | O_RDWR, 0666);
```
Comment (vlvkobal) [inline collectors/cgroups.plugin/sys_fs_cgroup.c:484]: Should it be world-readable/writable?

File: collectors/cgroups.plugin/sys_fs_cgroup.c [hunk 2]
```diff
@@ -2054,6 +2110,74 @@ static inline void copy_discovered_cgroups()
...
+    if (likely(stat(out->path, &buf) == 0)) {
+        return;
+    }
+
+    snprintfz(out->path, FILENAME_MAX, "%s%s/cgroup.procs", cgroup_devices_base, id);
+    if (likely(stat(out->path, &buf) == 0)) {
+        return;
+    }
+
+    snprintfz(out->path, FILENAME_MAX, "%s%s/cgroup.procs", cgroup_unified_base, id);
```
Comment (vlvkobal) [inline collectors/cgroups.plugin/sys_fs_cgroup.c:2137]: Why do you check the unified base if `cgroup_use_unified_cgroups=0`?

File: collectors/cgroups.plugin/sys_fs_cgroup.c (hunk 1 above)
Comment (thiagoftsm) [reply collectors/cgroups.plugin/sys_fs_cgroup.c:484]: This was used only for tests, I will remove it with next commit.

**Review by vlvkobal — APPROVED** [review]

**Review by UmanShahzad — APPROVED** [review]
Excellent work! I don't have any reservations.

**Pull request merged**